### DISCUSSION TITLE
可以出个paopao网关的docker镜像吗？

### DISCUSSION THREAD

**Discussion by gukey** [kkkgo/PaoPaoGateWay#112] [想法]
![adhemo&paopaodns](http://forumupload.ru/uploads/001b/b0/1b/8/332487.jpg)

主要是因为最近 [ROS-MikroTikPatch]( https://github.com/elseif/MikroTikPatch) 出破解版了，可以直接虚拟机安装ROS X86 破解版本，自带docker 功能，目前已经把paopaodns 和 adhome 部署在了ros上了，运行完美，就差个网关，如果有docker 网关镜像的话，就可以都部署在上面，少启用一个虚拟机，一个ros就全部搞定了，如果方便，可以考虑一下，谢谢；

**Comment by kkkgo** [comment]
参考：
https://github.com/kkkgo/PaoPaoGateWay/discussions/105

https://github.com/kkkgo/PaoPaoGateWay/discussions/57

https://github.com/kkkgo/PaoPaoGateWay/discussions/43#discussioncomment-7703022

docker网络环境较为复杂，具体支持到什么特性取决于宿主环境，除非每个环境（比如某某平台某某版本ROS）专门做适配和测试，不然无法做到通用，失去了docker镜像的意义。
另外docker作为网关大多数方案都是起tun接口，性能效率可能没这么高，如果你想要了解这方面的方案，你可以Google一下“docker tun fakeip ros网关”也许能找到对你有帮助的信息。

**Reply by gukey** [reply]
明白了，试用了几个calsh docker，效果没iso好！

**Reply by gukey** [reply]
![yacd](http://forumupload.ru/uploads/001b/b0/1b/8/998979.png)

还是没事折腾了一下，最终成功了，而且占用内存极小，可以翻墙，速度也不错，但有个问题就是 路由好像没有劫持到，默认是 7.0.0.0/8，ros 里面也加了静态路由，之前ping google.com，会被劫持到 7.0.0.X 地址，也能ping通，现在ping不通，翻墙正常，dns还是用的paopaodns，网关地址指向192.168.1.3，就是目前mihomo docker的IP，  这是我的mihomo配置，配置了FAKEIP的模式，麻烦看看是哪配置错误了，那些地方需要优化下；
```
####################################
# 创建者：v2rayse.com
# 客户端：Clash.Meta
# 创建时间：2024-08-11 19:58:58
# 使用模版：V2RaySE 默认模版
# 节点数量：1
####################################
# !!当前配置文件请使用基于 Clash.Meta 内核程序
mixed-port: 7890
allow-lan: true
log-level: info
mode: global
external-controller: 0.0.0.0:9090
dns:
  enable: true
  prefer-h3: true
  use-hosts: true
  use-system-hosts: true
  respect-rules: false
  listen: 0.0.0.0:53
  ipv6: false
  default-nameserver:
    - 223.5.5.5
  enhanced-mode: fake-ip
  fake-ip-range: 7.0.0.0/8
  fake-ip-filter:
    - '*.lan'
    - localhost.ptlogin2.qq.com
  nameserver-policy:
    'www.baidu.com': '114.114.114.114'
    '+.internal.crop.com': '10.0.0.1'
    'geosite:cn': https://doh.pub/dns-query
  nameserver:
    - https://doh.pub/dns-query
    - https://dns.alidns.com/dns-query
  fallback:
    - tls://8.8.4.4
    - tls://1.1.1.1
  proxy-server-nameserver:
    - https://doh.pub/dns-query
  fallback-filter:
    geoip: true
    geoip-code: CN
    geosite:
      - gfw
    ipcidr:
      - 240.0.0.0/4
    domain:
      - '+.google.com'
      - '+.facebook.com'
      - '+.youtube.com'

proxies:
  - {"name":"x-ui-08-3clqi4b1","type":"vless","server":"8.28.20.151","port":51514,"udp":true,"uuid":"168175a2-13f6-43ac-981f-44c6941e3f50","tls":true,"servername":"yahoo.com","flow":"xtls-rprx-vision","network":"tcp","reality-opts":{"public-key":"tP-FAmQCT9jTek4GLcnCjkNZB3iZmMKU-lo9w94-Dgg","short-id":"c75c041e"},"client-fingerprint":"random"}

tun:
  enable: true
  stack: system
  auto-route: true
  auto-detect-interface: true
  dns-hijack:
    - 7.0.0.0:53

**Reply by kkkgo** [reply]
这其实是正常的，ppgw考虑到ping和路由跟踪会把ICMP包劫持到本地，像tun接口没有什么额外处理的话ping不通也是正常的。
